File: coyote/models/fetch_from_stream_in.py
# this file is @generated
from pydantic import Field

from ..internal.base_model import BaseModel


class FetchFromStreamIn(BaseModel):
    batch_size: int = Field(alias="batch_size")
    """How many messages to read from the stream."""

    consumer_group: str = Field(alias="consumer_group")

    name: str

    visibility_timeout_seconds: int = Field(alias="visibility_timeout_seconds")
    """How long messages are locked by the consumer."""

File: src/models/fetch_from_stream_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct FetchFromStreamIn {
    /// How many messages to read from the stream.
    pub batch_size: u16,

    pub consumer_group: String,

    pub name: String,

    /// How long messages are locked by the consumer.
    pub visibility_timeout_seconds: u64,
}

impl FetchFromStreamIn {
    pub fn new(
        batch_size: u16,
        consumer_group: String,
        name: String,
        visibility_timeout_seconds: u64,
    ) -> Self {
        Self {
            batch_size,
            consumer_group,
            name,
            visibility_timeout_seconds,
        }
    }
}
